datafusion/physical-plan/src/coalesce_batches.rs

```diff
@@ -290,6 +294,46 @@ pub fn concat_batches(
     arrow::compute::concat_batches(schema, batches)
 }
 
+/// `StringViewArray` reference to the raw parquet decoded buffer, which reduces copy but prevents those buffer from being released.
+/// When `StringViewArray`'s cardinality significantly drops (e.g., after `FilterExec` or `HashJoinExec` or many others),
+/// we should consider consolidating it so that we can release the buffer to reduce memory usage and improve string locality for better performance.
+fn gc_string_view_batch(batch: &RecordBatch) -> RecordBatch {
+    let new_columns: Vec<ArrayRef> = batch
+        .columns()
+        .iter()
+        .map(|c| {
+            // Try to re-create the `StringViewArray` to prevent holding the underlying buffer too long.
+            if let Some(s) = c.as_string_view_opt() {
```

Comment:
A minor comment here is that I think you can reduce the level of nesting by using syntax like

```rust
let Some(s) = c.as_string_view_opt() else {
  return Arc::clone(c)
}
```